src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RazorComponentRenameEndpoint.cs

```diff
@@ -216,25 +217,45 @@ public async Task AddEditsForCodeDocument(List<WorkspaceEditDocumentChange> docu
                 Host = string.Empty,
                 Scheme = Uri.UriSchemeFile,
             }.Uri;
-            AddEditsForCodeDocument(documentChanges, originTagHelperBinding, newName, uri, codeDocument);
+            AddEditsForCodeDocument(documentChanges, originTagHelpers, newName, uri, codeDocument);
         }
 
-        public void AddEditsForCodeDocument(List<WorkspaceEditDocumentChange> documentChanges, TagHelperBinding originTagHelperBinding, string newName, DocumentUri uri, RazorCodeDocument codeDocument)
+        public void AddEditsForCodeDocument(List<WorkspaceEditDocumentChange> documentChanges, IReadOnlyList<TagHelperDescriptor> originTagHelpers, string newName, DocumentUri uri, RazorCodeDocument codeDocument)
         {
             var documentIdentifier = new VersionedTextDocumentIdentifier { Uri = uri };
             var tagHelperElements = codeDocument.GetSyntaxTree().Root
                 .DescendantNodes()
                 .Where(n => n.Kind == SyntaxKind.MarkupTagHelperElement)
                 .OfType<MarkupTagHelperElementSyntax>();
-            foreach (var node in tagHelperElements)
+
+            for (var i = 0; i < originTagHelpers.Count; i++)
```

Comment:
Is this just to avoid Enumerable allocation?

Reply:
Yes